src/classes/CRLP_Batch_Base_TEST.cls

```diff
@@ -0,0 +1,361 @@
+        configureUserDefinedInnerJoin(
+                CRLP_RollupProcessingOptions.RollupType.AccountHardCredit,
+                userDefinedLastNDaysFieldOverride,
+                userDefinedLastNDaysValueOverride);
+
+        CRLP_Account_BATCH accountBatchJob = new CRLP_Account_BATCH(new List<Id>());
+        accountBatchJob.initializeJobData();
+        accountBatchJob.isScheduledJob = true;
+
+        String generatedWhereClause = accountBatchJob.getMainQueryInnerJoinFilter();
+        System.assert(generatedWhereClause.contains('WHERE'), 'The incremental mode ' +
+                'where clause for the inner join should contain a where filter on the Opportunities.');
+        System.assert(generatedWhereClause.contains(userDefinedLastNDaysFieldOverride));
+        System.assert(generatedWhereClause.contains(String.valueOf(userDefinedLastNDaysValueOverride)));
+    }
+
+    @isTest
+    private static void contactHardCreditRollupWithUserDefinedJoinGeneratesProperJoinQuery() {
+        String userDefinedLastNDaysFieldOverride = 'LastModifiedDate';
+        Integer userDefinedLastNDaysValueOverride = 555;
+        configureUserDefinedInnerJoin(
+                CRLP_RollupProcessingOptions.RollupType.ContactHardCredit,
+                userDefinedLastNDaysFieldOverride,
+                userDefinedLastNDaysValueOverride);
+
+        CRLP_Contact_BATCH contactBatchJob = new CRLP_Contact_BATCH(new List<Id>());
+        contactBatchJob.initializeJobData();
+        contactBatchJob.isScheduledJob = true;
+
+        String generatedWhereClause = contactBatchJob.getMainQueryInnerJoinFilter();
+        System.assert(generatedWhereClause.contains('WHERE'), 'The incremental mode ' +
+                'where clause for the inner join should contain a where filter on the Opportunities.');
+        System.assert(generatedWhereClause.contains(userDefinedLastNDaysFieldOverride));
+        System.assert(generatedWhereClause.contains(String.valueOf(userDefinedLastNDaysValueOverride)));
+    }
+
+    /******************************** Helper Methods *****************************/
+
+    /**
+     * @description Activates or deactivates incremental mode for a RollupType.
+     * Used to test customized rollup configurations.
+     * @param jobType The RollupType whose incremental mode custom setting should be configured
+     * @param isIncrementalMode Whether the incremental mode settings should be activated (true)
+     * or deactivated (false)
+     * @return The configured Customizable_Rollup_Settings__c record
+     */
+    private static Customizable_Rollup_Settings__c configureIncrementalMode(
+            CRLP_RollupProcessingOptions.RollupType jobType,
+            Boolean isIncrementalMode) {
+
+        Customizable_Rollup_Settings__c settings;
+
+        switch on (jobType) {
+
+            when AccountHardCredit {
+                settings = UTIL_CustomSettingsFacade.getRollupSettingsForTests(
```

Comment:
@kslewis  I think the creation of the settings can be above the switch. Then within each switch you can set the appropriate field to true:
`settings.AccountHardCreditNonSkew_Incremental__c = isIncrementalMode;`

Reply:
Ok, updated!